### PR TITLE
Add support for multiple profiles

### PR DESCRIPTION
**Pull request by SchSeba**
This commit add the ability to use multiple profiles.

All the profiles with the same highest priority will be used

Signed-off-by: Sebastian Sch <sebassch@gmail.com>

### CODE REVIEW

**Review by fedepaol — COMMENTED** [review]
A few nits

File: pkg/controller/ptpconfig/recommend.go [hunk 1]
```diff
@@ -93,42 +85,79 @@ func getRecommendProfileName(
...
-			if len(r.Match) == 0 {
-				continue
+		// ignore if match section is empty
+		if len(r.Match) == 0 {
+			continue
+		}
+
+		// check if the policy match the node
+		if isMachNodeName(&node, r.Match) {
+			if !foundPolicy {
```
[the remark this answers is not in the document]
Comment (SchSeba) [reply pkg/controller/ptpconfig/recommend.go:108]: I didn't understand how you want to use a switch here can you explain please :)

**Comment by SchSeba** [comment]
@fedepaol I fix the comments except for the switch one.

can you please take another look?

**Comment by SchSeba** [comment]
/hold 

until 4.7 branch is open

File: pkg/controller/ptpconfig/recommend.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (fedepaol) [reply pkg/controller/ptpconfig/recommend.go:108]: Something like:
```
        switch {
            case !nodeMatches(&node, r.Match):
                continue
            case !foundPolicy:
                profilesNames[*r.Profile] = struct{}{}
				priority = *r.Priority
				foundPolicy = true
            case *r.Priority == priority:
   				profilesNames[*r.Profile] = struct{}{}
            default:
                break
        }
```

[the remark this answers is not in the document]
Comment (SchSeba) [reply pkg/controller/ptpconfig/recommend.go]: oh nice!

done

**Pull request merged**